assignment4/Q1.cpp

```diff
@@ -0,0 +1,126 @@
+#include <iostream>
+#include <vector>
+#include <set>
+
+using namespace std;
+//iterating two times - first to copy the board
+//not sure of its correctness
+//probably something is missing but I do not know yet
```

Comment:
Right, that is my algorithm. Good to know about "rank" system - I thought that you can implement union-set algorithm without it, thank you :) If this assignment was about right numbers for each cell of the island, I surely would implement it to do it correctly but I thought it is not necessary here - I'm glad that you agree. 
I have already read this article and few other about Union-find algorithms but I tried to find a way to simplify it. I am thinking about doing this assignment once again though - this time using disjoint-set properly. Thank you for all the comments :)